jumanji/wrappers.py

```diff
@@ -360,18 +360,40 @@ def render(self, state: State) -> Any:
         return super().render(state_0)
 
 
+OBS_IN_EXTRAS_KEY = "real_next_obs"
+
+
+def _obs_in_extras(
+    state: State, timestep: TimeStep[Observation]
+) -> Tuple[State, TimeStep[Observation]]:
+    """Place the observation in timestep.extras[OBS_IN_EXTRAS_KEY].
+    Used when auto-resetting to store the observation from the terminal TimeStep.
+
+    Args:
+        state: State object containing the dynamics of the environment.
+        timestep: TimeStep object containing the timestep returned by the environment.
+
+    Returns:
+        (state, timestep): where the observation is placed in timestep.extras[OBS_IN_EXTRAS_KEY].
+    """
+    extras = timestep.extras
+    extras[OBS_IN_EXTRAS_KEY] = timestep.observation
+    return state, timestep.replace(extras=extras)  # type: ignore
+
+
 class AutoResetWrapper(Wrapper):
```

Comment:
We also need to apply the extras next obs to the `VmapAutoResetWrapper`

Reply:
This is already done?

Reply:
Sorry, I didn't catch you applied it to both wrappers.